Lecture 10 Refactor

connortsui20 opened the pull request.

Fiona-CMU commented on week10/smart_pointers.md:
```diff
@@ -31,14 +31,57 @@ code {
+
+* `String`
+* `Vec<T>`
+* `Box<T>`
```
We haven't seen Box<T> at this point in the lecture?

connortsui20 replied:
They saw it in Week 8 (Ownership part 2)

Fiona-CMU reviewed (commented):
Reviewed first 34 slides. Really like the in-depth explanation of infinite size & making a clearer link between the smart pointers and trait objects. Everything in the first half looks good to me, with the exception of the one comment I left.

YHTerrance commented on week10/smart_pointers.md:
````diff
@@ -402,414 +623,316 @@ Here is the actual source code of `std::mem::drop` in the standard library:
----
+```rust
+fn main() {
+    let plan = String::new("Plan to steal Connor's identity, do not distribute");
````
This should be `String::from` instead of `String::new`

```suggestion
    let plan = String::from("Plan to steal Connor's identity, do not distribute");
```

connortsui20 commented:
I removed a lot and added a lot. This lecture is noticeably diluted, but I think that is okay. I'd like to get reviews on what is here right now, and if I think it is _too_ diluted then I will add some detail or add refcell back.
